python_modules/dagster/dagster/_core/definitions/assets.py

```diff
@@ -1255,21 +1256,14 @@ def update_replace_dict_and_conflicts(
             )
 
         check_specs_by_output_name = {
-            output_name: check_spec._replace(
-                asset_key=output_asset_key_replacements.get(
-                    check_spec.asset_key, check_spec.asset_key
-                )
+            output_name: check_spec.replace_key(
+                key=output_check_key_replacements.get(check_spec.key, check_spec.key)
             )
             for output_name, check_spec in self.node_check_specs_by_output_name.items()
         }
 
         selected_asset_check_keys = {
-            check_key._replace(
-                asset_key=output_asset_key_replacements.get(
-                    check_key.asset_key, check_key.asset_key
-                )
-            )
-            for check_key in self.check_keys
+            output_check_key_replacements.get(key, key) for key in self.check_keys
```

Comment:
so I think this is a breaking behavior change that has the potential to actually cause issues -- one good thing about the old implementation was that  if you're replacing the key of an asset, then you almost certainly need to change the key of any checks referencing that asset.

even though that is implicit and weird, I think it's easy for someone to end up relying on that behavior